### PR TITLE
chore | removing entities edge list formation in api trace graph

### PR DESCRIPTION
**Pull request by aman-bansal**

### CODE REVIEW

File: hypertrace-trace-enricher/hypertrace-trace-enricher-api/src/main/java/org/hypertrace/traceenricher/trace/util/GraphBuilderUtil.java [hunk 1]
```diff
@@ -23,9 +23,7 @@ static boolean isDifferentTrace(StructuredTrace cachedTrace, StructuredTrace tra
   }
 
   static boolean isStructuredTraceChanged(StructuredTrace cachedTrace, StructuredTrace trace) {
-    return isDifferentTrace(cachedTrace, trace)
-        || isTraceEventsChanged(cachedTrace, trace)
-        || isTraceEntitiesChanged(cachedTrace, trace);
+    return isDifferentTrace(cachedTrace, trace) || isTraceEventsChanged(cachedTrace, trace);
```
Comment (kotharironak) [inline hypertrace-trace-enricher/hypertrace-trace-enricher-api/src/main/java/org/hypertrace/traceenricher/trace/util/GraphBuilderUtil.java:26]: Do we build `TraceGraph` in multiple places or only ApiTraceGraph?

Comment (aman-bansal) [reply hypertrace-trace-enricher/hypertrace-trace-enricher-api/src/main/java/org/hypertrace/traceenricher/trace/util/GraphBuilderUtil.java:26]: It's being used in other places but does entities should have any change in trace graph? it should be same even if new entities are being added? correct

File: hypertrace-trace-enricher/hypertrace-trace-enricher-api/src/main/java/org/hypertrace/traceenricher/trace/util/StructuredTraceGraphBuilder.java [hunk 2]
```diff
@@ -17,14 +18,7 @@ public class StructuredTraceGraphBuilder {
...
     StructuredTraceGraph cachedGraph = cachedGraphThreadLocal.get();
-    boolean shouldRebuildTraceEventsGraph =
-        GraphBuilderUtil.isTraceEventsChanged(cachedTrace, trace);
-    boolean shouldRebuildTraceEntitiesGraph =
-        GraphBuilderUtil.isTraceEntitiesChanged(cachedTrace, trace);
-
-    if (null == cachedGraph
-        || GraphBuilderUtil.isDifferentTrace(cachedTrace, trace)
-        || (shouldRebuildTraceEventsGraph && shouldRebuildTraceEntitiesGraph)) {
+    if (null == cachedGraph || GraphBuilderUtil.isDifferentTrace(cachedTrace, trace)) {
```
Comment (kotharironak) [inline hypertrace-trace-enricher/hypertrace-trace-enricher-api/src/main/java/org/hypertrace/traceenricher/trace/util/StructuredTraceGraphBuilder.java:21]: I think, we need to use `isStructuredTraceChanged`, right?

File: hypertrace-trace-enricher/hypertrace-trace-enricher-api/src/main/java/org/hypertrace/traceenricher/trace/util/GraphBuilderUtil.java (hunk 1 above)
Comment (kotharironak) [reply hypertrace-trace-enricher/hypertrace-trace-enricher-api/src/main/java/org/hypertrace/traceenricher/trace/util/GraphBuilderUtil.java:26]: If there is no usage of `TraceEntitiesGraph` from `StructuredTraceGraph`, can we remove that private member from `StructuredTraceGraph`?

File: hypertrace-trace-enricher/hypertrace-trace-enricher-api/src/main/java/org/hypertrace/traceenricher/trace/util/StructuredTraceGraphBuilder.java [hunk 3]
```diff
@@ -39,65 +34,30 @@ public static StructuredTraceGraph buildGraph(StructuredTrace trace) {
...
-          (null != tempGraph.getTraceEntitiesGraph()));
-
-      tempGraph.reCreateTraceEntitiesGraph(trace);
-      LOG.info(
-          logPrefix
-              + "Recreating entities graph. Is events graph non-null: {}."
-              + " Is entities graph non-null: {}",
-          (null != tempGraph.getTraceEventsGraph()),
-          (null != tempGraph.getTraceEntitiesGraph()));
+          logPrefix + "Recreating events graph. Is events graph non-null: {}.",
```
Comment (kotharironak) [inline hypertrace-trace-enricher/hypertrace-trace-enricher-api/src/main/java/org/hypertrace/traceenricher/trace/util/StructuredTraceGraphBuilder.java:60]: Do we need this now - `reCreateTraceEventsGraph`? Same would have happen above, right?

**Review by kotharironak — DISMISSED** [review]
Minor comment for debug log, rest looks fine.

**Review by skjindal93 — APPROVED** [review]

**Review by kotharironak — APPROVED** [review]

**Pull request merged**